### PR TITLE
Wiggly lines for S waves in plot ray paths (addresses issue #3042)

### LINKED ISSUE #3042: Different plotting style for P- and S-waves
### Problem

It would be nice when plotting ray paths with `obspy.taup.plot_ray_paths()` to distinguish between P- and S-waves. As an example, see this Figure 4.15 from Shearer Introduction to Seismology that uses squiggles (sine waves) for S-waves. There could be other ways to do this too (e.g., different matplotlib line styles) Currently `plot_ray_paths` makes solid lines for all ray types.
raypaths

### Proposed solution

Modify the `plot_ray_paths` plotting code to distinguish between P- and S-waves, either with squiggly lines or different line styles.

### PR DESCRIPTION
**Pull request by johnrudge**
### What does this PR do?

This PR adds a flag `indicate_wave_type` (bool) to the `Arrivals.plot_rays()` method. When True the S waves are plotted as wiggly lines.

### Why was it initiated?  Any relevant Issues?

This fixes #3042.  The PR adds a function `split_ray_path` to `taup.utils` which is needed to split the path at discontinuities so that the P and S waves can be plotted with separate line styles. The `split_ray_path` function figures out whether the waves are P or S by looking at their delay time. There may well be more elegant ways of splitting the ray path, but the current method seems robust.

As for tests, I'm not quite sure how to test plotting features, so I've not added a test yet.

### PR Checklist
- [x] Correct base branch selected? `master` for new features, `maintenance_...` for bug fixes
- [ ] This PR is not directly related to an existing issue (which has no PR yet).
- [x] If the PR is making changes to documentation, docs pages can be built automatically.
      Just add the "build_docs" tag to this PR.
      Docs will be served at [docs.obspy.org/pr/{branch_name}](https://docs.obspy.org/pr/) (do not use master branch).
      Please post a link to the relevant piece of documentation.
- [ ] If all tests including network modules (e.g. `clients.fdsn`) should be tested for the PR,
      just add the "test_network" tag to this PR.
- [x] All tests still pass.
- [ ] Any new features or fixed regressions are covered via new tests.
- [x] Any new or changed features are fully documented.
- [x] Significant changes have been added to `CHANGELOG.txt` .
- [x] First time contributors have added your name to `CONTRIBUTORS.txt` .
- [ ] If the changes affect any plotting functions you have checked that the plots
      from all the CI builds look correct. Add the "upload_plots" tag so that plotting 
      outputs are attached as artifacts. 
- [ ] Add the "ready for review" tag when you are ready for the PR to be reviewed.

Code by myself and Stuart Russell, so I've added both our names to the CONTRIBUTORS.txt.
+DOCS

### CODE REVIEW

File: obspy/taup/tau.py [hunk 1]
```diff
@@ -246,6 +246,7 @@ def plot_times(self, phase_list=None, plot_all=True, legend=False,
 
     def plot_rays(self, phase_list=None, plot_type="spherical",
                   plot_all=True, legend=False, label_arrivals=False,
+                  indicate_wave_type=False,
                   show=True, fig=None, ax=None):
```
Comment (megies) [inline obspy/taup/tau.py:250]: Oh i wanted to mention that earlier.. can you please move the new kwarg to the end of the list? I know it logically fits better here, but this kinda isn't backwards compatible (in the highly unlikely case that somebody uses this with 6+ args ;)

**Comment by megies** [comment]
> Code by myself and Stuart Russell, so I've added both our names to the CONTRIBUTORS.txt.

:+1: totally fine =)

**Comment by megies** [comment]
> As for tests, I'm not quite sure how to test plotting features, so I've not added a test yet.

Image comparison tests arent run currently anyway, it's on our TODO to bring those back. We could simply have one of the plots in the Tutorial use this new option.

Maybe this one: https://docs.obspy.org/master/tutorial/code_snippets/travel_time.html#spherical-ray-paths

Or this last one on the page: https://docs.obspy.org/master/tutorial/code_snippets/travel_time.html#ray-paths-for-multiple-distances

Or maybe also on the taup front page: https://docs.obspy.org/master/packages/obspy.taup.html?highlight=taup#plotting

(see [here to change it in Tutorial](https://github.com/obspy/obspy/blob/master/misc/docs/source/tutorial/code_snippets/travel_time.rst) and [here for the taup API front page](https://github.com/obspy/obspy/blob/master/obspy/taup/__init__.py))

**Comment by megies** [comment]
Ideal would be if you rebase this on top of current master, to get rid of the merge commit and the duplication of changes. But, you'd have to fix some conflicts in the rebase manually with indentation changes involved and all, so if it's too much to ask, then it's fine too.

**Comment by trichter** [comment]
> I don't know exactly what's goin on, earlier I hit this button "workflows await approval" > "approve and run" so I thought all is in line (you can see github ran CI for an earlier commit), but now I get this shown to me again.. CC @trichter

Don't know either. Maybe its because, now another workflow (the docs building) needs to be started. Usually workflows need to be approved only once.

**Comment by johnrudge** [comment]
Thanks both! It says all check passed, but I can't see the documentation at https://docs.obspy.org/pr/
I locally built the docs on my machine and they look fine though. Next steps?

**Comment by trichter** [comment]
Docs deployment is working again. It was not working since 4 days, because the value of the `status` field changed from `"success"` to `"completed"`. I adapted the server-side script.

```
curl -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/obspy/obspy/actions/runs?per_page=1
{
  "total_count": 1844,
  "workflow_runs": [
    {
      "id": 2155599922,
      "name": "docs",
      "node_id": "WFR_kwLOAFd0FM6Ae9gy",
      "head_branch": "improve-plot-ray-paths",
      "head_sha": "86de2de47fd138f703148965bf93c807decd9ffc",
      "run_number": 507,
      "event": "pull_request",
      "status": "completed",
      "conclusion": "success",

...
```

**Comment by megies** [comment]
> Next steps?

Nothing much, I'll give it a final glance in a bit, then merge. =)

https://docs.obspy.org/pr/improve-plot-ray-paths/tutorial/code_snippets/travel_time.html#spherical-ray-paths

Looking beautiful in docs!

https://docs.obspy.org/pr/improve-plot-ray-paths/tutorial/code_snippets/travel_time_body_waves.png

travel_time_body_waves


> the value of the `status` field in the answer to the following request changed from `"success"` to `"completed"`.

Euh! Thanks for finding and fixing it =)

**Comment by johnrudge** [comment]
Thanks for reviewing and merging it so quickly!

**Comment by QuLogic** [comment]
> > I don't know exactly what's goin on, earlier I hit this button "workflows await approval" > "approve and run" so I thought all is in line (you can see github ran CI for an earlier commit), but now I get this shown to me again.. CC @trichter
> 
> Don't know either. Maybe its because, now another workflow (the docs building) needs to be started. Usually workflows need to be approved only once.

Workflows for new contributors must be approved every time, i.e., for every new push.

**Comment by jrekoske** [comment]
@johnrudge Thanks for getting to my issue so quickly! This is a really neat addition to obspy.

**Pull request merged**